lib/ASRouter.jsm

```diff
@@ -307,12 +319,33 @@ class _ASRouter {
     }
   }
 
+  _validPreviewEndpoint(url) {
+    try {
```

Comment:
Can you maybe throw/log appropriate errors that output useful info (e.g. `The preview URL host ${endpoint.host} is not in the whitelist`) for debugging purposes?

Reply:
I went with logging just `url`, I'm assuming the value might be invalid and constructing the URL object can fail/throw.